pkgs/by-name/ad/addwater/package.nix

```diff
@@ -0,0 +1,56 @@
+{
+  blueprint-compiler,
+  desktop-file-utils,
+  fetchFromGitHub,
+  lib,
+  libadwaita,
+  libportal-gtk4,
+  meson,
+  ninja,
+  python3Packages,
+  wrapGAppsHook4,
+  appstream,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "addwater";
+  version = "1.1.3";
+  # built with meson, not a python format
+  format = "other";
+
+  src = fetchFromGitHub {
+    owner = "largestgithubuseronearth";
+    repo = "addwater";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-TUNDUJxYM9v1UECYSC+jJ+53hFGPChivz9kqyU9Qee4=";
+  };
+
+  buildInputs = [
+    libadwaita
+    libportal-gtk4
+  ];
+
+  nativeBuildInputs = [
+    blueprint-compiler
+    desktop-file-utils
+    meson
+    ninja
+    wrapGAppsHook4
+    appstream
```

Comment:
No `gobject-introspection`? Are you sure it can launch?

Reply:
Yes, but I found out that the settings don't start because they use AdwButtonRow. However, AdwButtonRow is only available from libadwaita version 1.6.